packages/npm-dependencies-validation/src/index.ts

```diff
@@ -0,0 +1,68 @@
+import { spawnCommand } from "./utils/npmUtil";
+import { resolve, dirname } from "path";
+import { isManagedByNpm6 } from "./utils/packageJsonUtil";
+import {
+  DependencyIssue,
+  IssueType,
+  NpmLsDependency,
+  VscodeFsUri,
+} from "./types";
+
+const LS_DEPS = ["ls", "--depth=0"]; // dependencies and extraneous modules
+const LS_DEV_DEPS = [...LS_DEPS, "--dev"]; // devDependencies
+
+async function getDepsStatus(
+  packageJsonPath: string,
+  devDeps: boolean
+): Promise<any> {
+  // try to use DependencyType ???
+  return spawnCommand(
+    devDeps ? [...LS_DEV_DEPS] : [...LS_DEPS],
+    resolve(dirname(packageJsonPath))
+  );
+}
+
+function getProblemType(dependency: NpmLsDependency): IssueType | undefined {
+  if (dependency.missing) return "missing";
+  if (dependency.invalid) return "invalid";
+  if (dependency.extraneous) return "extraneous";
+}
+
+async function getIssues(
+  path: string,
+  devDeps = false
```

Comment:
- config object?
- I don't think we want a default value for `devDeps` as it can add a branch that needs testing for 100% coverage.
-

Reply:
Done